lib/matplotlib/tests/test_axes.py

```diff
@@ -1517,6 +1517,22 @@ def test_pcolorargs():
         ax.pcolormesh(X, Y, Z, shading='auto')
 
 
+def test_pcolormesh_underflow_error():
+    """
+    Test that underflow errors don't crop up in pcolormesh.  Probably
+    a numpy bug, but...
+    """
+    oldsettings = np.geterr()
+    np.seterr(under="raise")
```

Comment:
Worth using the [`np.errstate` context manager](https://numpy.org/doc/stable/reference/generated/numpy.errstate.html) instead to save a couple of lines?